entities/weapons/weaponchecker/shared.lua

```diff
@@ -127,19 +127,20 @@ end
 
 function SWEP:GetStrippableWeapons(ent, callback)
     CAMI.PlayerHasAccess(ent, "DarkRP_GetAdminWeapons", function(access)
-        for k,v in pairs(ent:GetWeapons()) do
-            if not v:IsValid() then continue end
-            local class = v:GetClass()
+        for _,w in pairs(ent:GetWeapons()) do
+            if not w:IsValid() then continue end
+            local class = w:GetClass()
+            local illegal = hook.Call("weaponIllegal", nil, w)
 
             if GAMEMODE.Config.weaponCheckerHideDefault and (table.HasValue(GAMEMODE.Config.DefaultWeapons, class) or
                 access and table.HasValue(GAMEMODE.Config.AdminWeapons, class) or
-                ent:getJobTable() and ent:getJobTable().weapons and table.HasValue(ent:getJobTable().weapons, class)) then
+                ent:getJobTable() and ent:getJobTable().weapons and table.HasValue(ent:getJobTable().weapons, class) or illegal) then
```

Comment:
This should probably be in its own `if` check as `GAMEMODE.Config.weaponCheckerHideDefault` should have no affect on the hook.

Also, would it not make more sense if a return value of `true` meant that it _can_ be stripped since the hook is called `weaponIllegal`?